OpenRA.Game/Activities/Activity.cs

```diff
@@ -9,50 +9,218 @@
  */
 #endregion
 
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using OpenRA.Traits;
 
 namespace OpenRA.Activities
 {
+	public enum ActivityState { Queued, Active, Done, Canceled }
+
 	public abstract class Activity
 	{
-		public Activity NextActivity { get; set; }
+		public ActivityState State { get; private set; }
+
+		public Activity Root
+		{
+			get
+			{
+				if (Parent == null)
+					return this;
+
+				var parent = Parent;
```

Comment:
What is the purpose of using `Parent` above, then shadowing the private `parent` member here?